cleaner/mseCleaner.go

```diff
@@ -0,0 +1,272 @@
+			}
+			rate.NO = strings.TrimSpace(word[0])
+			rate.HIGH = strings.TrimSpace(word[1])
+			rate.LOW = strings.TrimSpace(word[2])
+			rate.CODE = strings.TrimSpace(word[3])
+			rate.BUY = parseFloat(strings.TrimSpace(word[4]))
+			rate.SELL = parseFloat(strings.TrimSpace(word[5]))
+			rate.PCP = parseFloat(strings.TrimSpace(word[6]))
+			rate.TCP = parseFloat(strings.TrimSpace(word[7]))
+			rate.VOL = parseFloat(strings.TrimSpace(word[8]))
+			rate.DIVNET = parseFloat(strings.TrimSpace(word[9]))
+			rate.DIVYIELD = parseFloat(strings.TrimSpace(word[10]))
+			rate.EEARNYIELD = parseFloat(strings.TrimSpace(word[11]))
+			rate.PERATIO = parseFloat(strings.TrimSpace(word[12]))
+			rate.PBVRATION = parseFloat(strings.TrimSpace(word[13]))
+			rate.CAP = parseFloat(strings.TrimSpace(word[14]))
+			rate.PROFIT = parseFloat(strings.TrimSpace(word[15]))
+			rate.SHARES = parseFloat(strings.TrimSpace(word[16]))
+			rates = append(rates, rate)
+		}
+	}
+
+	if len(errors) > 0 {
+		affected := fmt.Sprintf("File: %q", csvFile)
+		errors = append([]string{affected}, errors...)
+		logErrors(errors, errorPath, date)
+	}
+
+	func() {
+		file, err := os.Create(fmt.Sprintf("%s%s.csv", cleanCSVPath, strings.ReplaceAll(date, "/", "-")))
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer file.Close()
+		w := csv.NewWriter(file)
+		for _, rate := range rates {
+			if err := w.Write([]string{
+				rate.NO, rate.HIGH, rate.LOW, rate.CODE,
+				fmt.Sprint(rate.BUY), fmt.Sprint(rate.SELL), fmt.Sprint(rate.PCP), fmt.Sprint(rate.TCP), fmt.Sprint(rate.VOL),
+				fmt.Sprint(rate.DIVNET), fmt.Sprint(rate.DIVYIELD), fmt.Sprint(rate.EEARNYIELD),
+				fmt.Sprint(rate.PERATIO), fmt.Sprint(rate.PBVRATION), fmt.Sprint(rate.CAP), fmt.Sprint(rate.PROFIT), fmt.Sprint(rate.SHARES)}); err != nil {
+				log.Fatalln("error writing record to csv:", err)
+			}
+		}
+
+		w.Flush()
+
+		if err := w.Error(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	fmt.Println("Finished file: ", csvFile)
+
+	return CleanedData{
+		dailyRates: rates,
+		date:       date,
+		errors:     errors,
+	}
+}
+
+func logErrors(errors []string, path string, date string) {
+	f, err := os.Create(fmt.Sprintf("%s%s-error.txt", path, strings.ReplaceAll(date, "/", "-")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	_, err2 := f.WriteString(strings.Join(errors, "\n"))
+	if err2 != nil {
+		log.Fatal(err)
+	}
+}
+
+func parseFloat(value string) float64 {
+	if strings.Contains(value, ",") {
+		value = strings.ReplaceAll(value, ",", "")
+	}
+	if strings.Contains(value, "(") {
+		raw := strings.ReplaceAll(value, "(", "")
+		raw = strings.ReplaceAll(raw, ")", "")
+
+		raw = strings.TrimSpace(raw)
+		value = fmt.Sprint("-", raw)
+	}
+
+	f, err := strconv.ParseFloat(value, 64)
+	if err != nil {
+		fmt.Println(err)
+		return 0
+	}
+	return f
+}
+
+func Verify(rate []string) (bool, error) {
+	var err []string
+	isValid := true
+	for i, value := range rate {
+		if i > 2 {
+			if isEmpty(value) {
```

Comment:
Watch out for nested if statement. Use a compound conditional instead

Reply:
okay, will switch to compound conditionals